### ISSUE TITLE
Not able to convert model to c array after quantization

### ISSUE THREAD

**Issue by juliowissing-iis** [tensorflow/tensorflow#44173]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution Ubuntu 20.04:
- TensorFlow installed from source:
- Tensorflow version 2.3:
- Target platform Arduino Nano 33

**Describe the problem**
I am optimizing a model for the Arduino Nano 33. After training I quantized the model using the tf lite converter. The resulting tf-lite file is as expected much smaller, but I am not able convert it now using the xxd -i command. The non quantized model however, does work as expected and I am able to run inference with it on the microcontroller.

**Please provide the exact sequence of commands/steps when you ran into the problem**
I trained a functional keras model with tf 2.3 and gpu acceleration. After that I used the tf-lite converter class to quantize the model with my validation set and also did a conversion without quantization. Subsequently, I used the xxd -i command to convert the tflite files to a .cc file. The one without quantization looks as expected and I can copy it into my arduino code. The quantized model on the other hand is not a byte array. The file is made of lines like this:

```
00000000: 2400 0000 5446 4c33 0000 0000 0000 0000  $...TFL3........
00000010: 0000 1200 1c00 0400 0800 0c00 1000 1400  ................
00000020: 0000 1800 1200 0000 0300 0000 fc17 0000  ................
00000030: 7004 0000 5804 0000 3c00 0000 0400 0000  p...X...<.......
00000040: 0100 0000 0c00 0000 0800 0c00 0400 0800  ................
00000050: 0800 0000 0800 0000 1a00 0000 1300 0000  ................
00000060: 6d69 6e5f 7275 6e74 696d 655f 7665 7273  min_runtime_vers
```

Can anyone provide information to this problem?

**Comment by juliowissing** [comment]
Short update: When quantizing the model with dynamic quantization (converter.optimizations = [tf.lite.Optimize.DEFAULT]), I am able to convert it to a c array with xxd. However, I can not load it onto my arduino, probably because mixed datatypes are not supported by the microcontroler?

**Comment by gowthamkpr** [comment]
@juliowissing-iis You can include your model as a C array following the procedure [here](https://www.tensorflow.org/lite/microcontrollers/build_convert#convert_to_a_c_array). Please look into it and let me know if you are still running into any issues. Thanks!
